src/JointTrajectoryControllerBase.cpp

```diff
@@ -297,8 +311,22 @@ void JointTrajectoryControllerBase::updateStep(const ros::Time& time,
 
   for (size_t idof = 0; idof < mAdapters.size(); ++idof)
   {
-    mAdapters[idof]->update(time, period, mActualPosition[idof],
-                            mDesiredPosition[idof], mActualVelocity[idof],
+    // Check for SO2
+    auto actualPos = mActualPosition[idof];
+    auto desiredPos = mDesiredPosition[idof];
+    auto jointSpace = mControlledSpace->getJointSpace(idof);
+    auto r1Joint = std::dynamic_pointer_cast<const R1Joint>(jointSpace);
+    if(!r1Joint) {
+      if(desiredPos - actualPos > M_PI) {
```

Comment:
Same comment as above. How do you guarantee that doing one round of +/-2pi is sufficient?

Reply:
It is safer to have the `std::fmod()` but wouldn't the discrepancy only come within one round of +/-2pi? Since `mDesiredPosition` starts between -pi and pi while `mActualPosition` start between 0 and 2pi, the discrepancy should be only within 2pi. Any additional rounds are accounted by `mCurrentTrajectoryOffset`.

Reply:
I wonder if we need `mCurrentTrajectoryOffset` anymore when fmoding the difference

Reply:
Could be something to clean up in another PR. But I'll just leave fmod here for this one.